src/utils.cpp

```diff
@@ -150,6 +150,30 @@ void OpenGLDebugMsg(const char *format, ...)
 	fputs(buf, stderr);
 }
 
+std::string format_duration(double seconds)
+{
+	std::ostringstream ss;
+	int duration = seconds;
+	int secs = duration % 60;
+	int minutes = (duration / 60) % 60;
+	int hours = (duration / 60 / 60) % 24;
+	int days = (duration / 60 / 60 / 24) % 7;
+	int weeks = (duration / 60 / 60 / 24 / 7);
+	if(weeks != 0)
```

Comment:
you could use `minutes` to compute `hours`, `hours` to compute `days`, etc. I don't care which is which and what is the better approach. Just a though.